source/include/core_json.h

```diff
@@ -116,21 +118,20 @@ JSONStatus_t JSON_Validate( const char * buf,
  * @note JSON_Search() performs validation, but stops upon finding a matching
  * key and its value. To validate the entire JSON document, use JSON_Validate().
  *
- * @return #JSONSuccess if the queryKey is found and the value output;
+ * @return #JSONSuccess if the query is matched and the value output;
  * #JSONNullParameter if any pointer parameters are NULL;
- * #JSONBadParameter if the queryKey is empty, or any subpart is empty, or max is 0;
- * #JSONIllegalDocument if the buffer contents are NOT valid JSON;
- * #JSONMaxDepthExceeded if object and array nesting exceeds a threshold;
- * #JSONNotFound if the queryKey is NOT found.
+ * #JSONBadParameter if the query is empty, or the portion after a separator is empty,
+ * or max is 0, or an index is too large to convert;
```

Comment:
Improve clarity about cause of error from passed index in query string
```suggestion
 * or max is 0, or an index is too large to convert to an integer value;
```

Reply:
changed